ui/src/app/components/webcomponents/test-plan-device-form.component.ts

```diff
@@ -102,8 +102,14 @@ export class TestPlanDeviceFormComponent extends TestPlanPlatformOsVersionFormCo
   setAgentDevice(agentDevice: AgentDevice) {
     this.agentDevice = null;
     this.agentDevice = agentDevice;
-    this.platformOsVersion=this.platformOsVersions.find(osVersion => osVersion.version==agentDevice.osVersion+".0")
-    this.environmentFormGroup.controls['platformOsVersionId'].setValue(this.platformOsVersion.id)
+    this.platformOsVersions;
+    this.platformService.findAllOsVersions(this.platform,this.version.workspace.workspaceType,this.testPlanLabType).subscribe(res => {
+      if(res){
+        this.platformOsVersions = res
+        this.platformOsVersion=this.platformOsVersions.find(osVersion => osVersion.version==agentDevice.osVersion+".0")
```

Comment:
Should os version always end with ".0" ? Is it tested ?

Reply:
yes OS version have to end with ".0"